plugins/notify.py

```diff
@@ -6,21 +6,38 @@ class NotifyPlugin(WillPlugin):
 
     @route("/notify/<build_id>/<text>")
     def send_notification(self, build_id, text):
-        notification_list = self.load("notify_{}".format(build_id), '')
-        self.say("{} BuildID: {}, Message: {}".format(notification_list, build_id, text)) 
+        if '@' in build_id:
+            notification_list = build_id
+        else:
+            notification_list = self.load("notify_" + build_id, {})
+            for room in notification_list:
+                self.say("{} BuildID: {}, Message: {}".format(notification_list[room], build_id, text), room=self.get_room_from_name_or_id(room)) 
 
-    @respond_to("^subscribe (@?)(?P<user>\w*) to (?P<build_id>\w*)")
+    @respond_to("^subscribe (@?)(?P<user>\S+) to (?P<build_id>\S+)")
     def subscribe(self, message, user, build_id):
         """
         subscribe [user] to [buildid]: request to be notified when builds complete
         """
         if user == "me":
             user = message.sender.nick
+        channel = self.get_room_from_message(message)['name']
+        notification_list = self.load("notify_" + build_id, {})
+        notification_list[channel] = "{} @{}".format(notification_list.get(channel, ""), user)
```

Comment:
I would say either make `notification_list[channel]` a list, or use `+=` here instead of `format`.